provider/pkg/await/deployment.go

```diff
@@ -150,34 +150,72 @@ func (dia *deploymentInitAwaiter) Await() error {
 		return err
 	}
 
-	// Create Deployment watcher.
-	deploymentWatcher, err := deploymentClient.Watch(context.TODO(), metav1.ListOptions{})
+	// Create Deployment watcher starting from the deployment's resource version, if set.
+	var depListOptions metav1.ListOptions
+	if dia.deployment != nil {
+		depListOptions.ResourceVersion = dia.deployment.GetResourceVersion()
+	}
+	deploymentWatcher, err := deploymentClient.Watch(context.TODO(), depListOptions)
```

Comment:
Probably, a naive question, but why do we start watching for deployments if `dia.deployment` is nil? (same applies to others)

Reply:
No you are right. I think we can just assume a deployment exists in the Await case. Removed the check here.